zstd/seqdec.go

```diff
@@ -101,6 +101,10 @@ func (s *sequenceDecs) initialize(br *bitReader, hist *history, out []byte) erro
 // execute will execute the decoded sequence with the provided history.
 // The sequence must be evaluated before being sent.
 func (s *sequenceDecs) execute(seqs []seqVals, hist []byte) error {
+	if len(hist) == 0 && len(s.dict) == 0 {
+		return s.executeSimple(seqs)
+	}
```

Comment:
> Let's focus on getting history in. I can take a stab looking at the combined decoder+execute, since this should be enough to make `decodeSync` working - it never has history IIRC.

Please ping me on wojciech_mula@poczta.onet.pl, I'll send you my current solution of `decodeSync`. It's mainly based on what we already merged. Of course, it would be perfectly fine if you left this job for me. :)

Reply:
@WojciechMula Actually, it will be a while before I can look at it.

I will ping you before I start anything 👍🏼